### ISSUE TITLE
Unhandled Promise Rejection: Native method execution error occurred

### ISSUE THREAD

**Issue by aianshume** [neutralinojs/neutralinojs#607]
**Expected Behavior**  
`status: false` this file is not exist
  
**Actual Behavior**  
`Unhandled Promise Rejection: Native method execution error occurred`

**Steps to Reproduce the Problem**  
this is the function which i created and when i call the function then its show the error

```
export const checkForDependancy = async() => {
  let response = await window.Neutralino.filesystem.getStats({
    path: './sampleVideo.mp4'
  });
  console.log('Stats:', response);
}
```

**Specifications**  
--- Global ---
Neutralinojs binaries: v2.8.0
Neutralinojs client: v1.5.0
neu CLI: v6.2.0

system: xubuntu 20.04

**Comment by Blatman** [comment]
This also occurs on MacOS (10.13.6) with v2.80/1.50
I use getsStats() to detect a folders existence and if it errors then I create the folder however it always errors as above even if the folder exists and then the createFolder says it can't create a folder within the tested folder.
It does not appear to detect the existence of either file or folder.

**Comment by tomaspinter** [comment]
already reported this a few days back (cannot find ticket any more); I have been advised to use url instead of path but that does not work either (running same neu version on Ubuntu 18.04.6)

**Comment by Blatman** [comment]
@tomaspinter - found the ticket in the client lib repository and I had the same results - I have yet to find an example of getStats that works if the file/folder already exists - hopefully the rehash of the API just done will improve this.

**Comment by shalithasuranga** [comment]
Hey.. all I just test and it worked like this on Ubuntu 
![image](https://user-images.githubusercontent.com/18402702/134761538-64b4dbb4-9e85-4154-bc26-078db176740a.png)

Not sure how you got the run time error instead. There are two possible things to do I think.

1. Throwing (Rejecting the promise) an error code with the error if the path doesn't exist. Eg: {error: {message: "Doesn't exist....", code: "ENOENT"}}
2. Resolving the promise with {exists: false} 

Which one do you think is good ?

**Comment by tomaspinter** [comment]
...ah, like this:

`await Neutralino.filesystem.getStats('./path/to/my/file/or/dir')`

it works indeed! For some reason the main thing I took from my previous ticket was the word 'url' so I was still trying it like this:

```
await window.Neutralino.filesystem.getStats({
    path: 'file:///path/to/my/file'
  });
```

My bad, sorry for that. Thanks a lot for clarifying. Would be maybe good to update the sample getStats code snippet in the documentation at https://neutralino.js.org/docs/api/filesystem so that users are aware

**Comment by Blatman** [comment]
Yes indeed if path strings are used it works but objects (as per the API Docs snippets) will fail. I have found this also applies to the copyFile and moveFile commands as well as getStats.

**Comment by Blatman** [comment]
Maybe this could be closed as addressed in 3.0.0 and docs updated.

**Comment by shalithasuranga** [comment]
Hey.. Thanks for notifying about this.. yeah we can close this one :)
